services/workers/src/main/java/org/sagebionetworks/table/worker/MaterializedViewUpdateWorker.java

```diff
@@ -1,23 +1,60 @@
 package org.sagebionetworks.table.worker;
 
+import org.apache.logging.log4j.Logger;
+import org.sagebionetworks.LoggerProvider;
 import org.sagebionetworks.asynchronous.workers.changes.ChangeMessageDrivenRunner;
 import org.sagebionetworks.common.util.progress.ProgressCallback;
+import org.sagebionetworks.repo.manager.table.MaterializedViewManager;
+import org.sagebionetworks.repo.manager.table.TableIndexConnectionUnavailableException;
+import org.sagebionetworks.repo.model.ObjectType;
+import org.sagebionetworks.repo.model.entity.IdAndVersion;
+import org.sagebionetworks.repo.model.jdo.KeyFactory;
 import org.sagebionetworks.repo.model.message.ChangeMessage;
+import org.sagebionetworks.repo.model.message.ChangeType;
+import org.sagebionetworks.repo.model.table.TableUnavailableException;
 import org.sagebionetworks.workers.util.aws.message.RecoverableMessageException;
+import org.sagebionetworks.workers.util.semaphore.LockUnavilableException;
+import org.springframework.beans.factory.annotation.Autowired;
 
 /**
  * This worker will build/rebuild the index for materialized views.
  *
  */
 public class MaterializedViewUpdateWorker implements ChangeMessageDrivenRunner {
 
+	private final MaterializedViewManager materializedViewManager;
+	private final Logger log;
+
+	@Autowired
+	public MaterializedViewUpdateWorker(MaterializedViewManager materializedViewManager,
+			LoggerProvider loggerProvider) {
+		super();
+		this.materializedViewManager = materializedViewManager;
+		this.log = loggerProvider.getLogger(MaterializedViewUpdateWorker.class.getName());
```

Comment:
instead of doing this in the constructor, you can create a method for initialization, e.g. public void configure that is annotated with @Autowired.